src/Front/DesugarLists.hs

```diff
@@ -26,4 +27,10 @@ desugarLists (Program decls) = return $ Program $ desugarTop <$> decls
 --     (App (App (Id "Cons") (Lit (LitInt 2))) 
 --          (App (App (Id "Cons") (Lit (LitInt 3))) (id "Nil")))
 desugarExpr :: Expr -> Expr
-desugarExpr e = e -- TODO: actually implement transformation
+desugarExpr (ListExpr es) = func es
+desugarExpr e = e 
```

Comment:
Looks good!
Replace lines 21 and 22 with
```
  desugarDef (DefFn v bs t e) = DefFn v bs t $ everywhere (mkT desugarExpr) e
  desugarDef (DefPat b e    ) = DefPat b $ everywhere (mkT desugarExpr) e
```
You will likely need to import Data.Generics to make `everywhere` and `mkT` work.